### ISSUE TITLE
[Grid] SessionMap backed by JDBC

### ISSUE THREAD

**Issue by diemol** [SeleniumHQ/selenium#8172]
## 🚀 Feature Proposal

Right now all the SessionMap information is stored in memory.

## Motivation

It'd be nice to have a SessionMap backed up by JDBC to enable more running instances of a SessionMap

**Comment by raju249** [comment]
Hey @diemol 

I will be working on this issue. I have setup the project locally successfully. 

I have a few questions as mentioned below:

1. I looked at the codebase particularly [this directory](https://github.com/SeleniumHQ/selenium/tree/master/java/server/src/org/openqa/selenium/grid/sessionmap) and I think what we intend to do in this issue is add a support for JDBC backed session map, similar to others already done here like `RedisBacked`, `remote` and `local`. Is that right?

2. For doing JDBC, we would need DB url of format 
`jdbc:<db_engine>://<db_url>:<db_port>/<database_name>`, `username` and `password` for database connection. So I think we can accept this as options from `SessionMapOptions.java`. I can see the `host` option for `RedisBacked` implementation. What do you think of adding new options for `DB url, username and password` for JDBC implementation or prefer some other way?

For point 2, if we add new options, I think we would need to add them to `SessionMapFlags.java` too for letting user see the new flags. Is this right?

Please let me know what you think of these points 🙇 
Thanks 🙏

**Comment by shs96c** [comment]
You're right: we're doing something analogous to the `RedisBackedSessionMap`, but instead of Redis, we'd be using JDBC.

The entry point is the method with the signature `public static SessionMap create(Config)`. You can create new classes annotated with `HasRoles` (compile with a dependency on `"//java:auto-service"` and a service file will be created automatically) and the `Main` class will add any flags declared on them automatically. These classes are typically named with a class ending in `Flags`. This allows you to have a specific `--jdbc-url` flag.

It's probably best to throw an exception if the tables required aren't present, rather than assuming some defaults. If your message contains information about what went wrong, then users have a chance to fix it. If you feel that more information is helpful, you can extend the `info` command.

In summary, I think you need at least two classes: `JdbcBackedSessionMap` and `JdbcSessionMapFlags`
